### PR TITLE
Rework EnumConverter.

### PR DESCRIPTION
**Pull request by mikelehen**
This avoids generics (and is perhaps a bit simpler).  The key difference is that I use `Enum.ToObject()` to convert the numeric value to the TargetType enum instead of having to use generics and a cast.  But I also ended up reworking the whole class to get rid of the `PrimitiveConverter` classes since it seemed simpler to just use a couple `switch` / `case` statements now.

Context: I had to rework the fork of this code that we're using in the Unity SDK due to Unity / iOS limitations.

I don't actually care too much if you want to accept this PR or not but I wanted to at least see if you have any thoughts / feedback on the approach.

### CODE REVIEW

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs [hunk 1]
```diff
@@ -26,94 +26,65 @@ namespace Google.Cloud.Firestore.Converters
     /// </summary>
     internal sealed class EnumConverter : ConverterBase
     {
-        private static readonly ConcurrentDictionary<BclType, PrimitiveConverter> s_primitiveConverters =
-            new ConcurrentDictionary<BclType, PrimitiveConverter>();
-
-        private readonly PrimitiveConverter _converter;
+        private TypeCode typeCode;
```
Comment (jskeet) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs]: Could you rename this to `_typeCode` and make it readonly please?

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs [hunk 2]
```diff
@@ -26,94 +26,65 @@ namespace Google.Cloud.Firestore.Converters
...
+        }
 
-            private sealed class Int64EnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => (long) value;
-                internal override object Int64ToEnum(long value) => (T) (object) value;
+        public override Value Serialize(SerializationContext context, object value) =>
+            new Value { IntegerValue = EnumToInt64(value) };
+
+        private long EnumToInt64(object value) {
```
Comment (jskeet) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs]: Please fix the formatting to "brace at start of line" here and in the next line.
For the cases, I'm fine with the existing formatting, or the return statement could be on the same line as the case.

Comment (jskeet) [reply apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs]: (And all of this for Int64ToEnumBaseType too :)

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs (hunk 1 above)
Comment (mikelehen) [reply apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs]: Woops, good catch.

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs (hunk 2 above)
Comment (mikelehen) [reply apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs]: Done.  The return statements on same line look nice.  Thanks.

**Pull request merged**